src/main/java/com/accenture/application/utils/ZipUtils.java

```diff
@@ -0,0 +1,185 @@
+
+        try {
+
+            in = new ZipInputStream(new BufferedInputStream(new FileInputStream(zipName)));
+            @SuppressWarnings("unused")
+			ZipEntry entry;
+            while ((entry = in.getNextEntry()) != null) {
+                int count;
+                byte data[] = new byte[1024];
+
+                out = new BufferedOutputStream(new FileOutputStream(fileOut.getAbsolutePath()), 1024);
+                while ((count = in.read(data, 0, 1024)) != -1) {
+                    out.write(data, 0, count);
+                }
+                out.flush();
+                IOUtils.closeQuietly(out, true);
+            }
+
+            return fileOut;
+        } catch (Exception e) {
+            logger.error(e.getMessage(), e);
+            return null;
+        } finally {
+            IOUtils.closeQuietly(out, true);
+            IOUtils.closeQuietly(in, true);
+        }
+    }
+
+    public File unzipFile(InputStream stream, File fileOut)
+            throws FileNotFoundException {
+
+        BufferedOutputStream out = null;
+        ZipInputStream in = null;
+
+        try {
+
+            in = new ZipInputStream(new BufferedInputStream(stream));
+            @SuppressWarnings("unused")
+			ZipEntry entry;
+            while ((entry = in.getNextEntry()) != null) {
+                int count;
+                byte data[] = new byte[1024];
+
+                out = new BufferedOutputStream(new FileOutputStream(fileOut.getAbsolutePath()), 1024);
+                while ((count = in.read(data, 0, 1024)) != -1) {
+                    out.write(data, 0, count);
+                }
+                out.flush();
+                IOUtils.closeQuietly(out, true);
+            }
+
+            return fileOut;
+
+        } catch (Exception e) {
+            logger.error(e.getMessage(), e);
+            return null;
+        } finally {
+            IOUtils.closeQuietly(out, true);
+            IOUtils.closeQuietly(in, true);
+        }
+    }
+
+    public void unzipFilesToDir(String zipName, File outputDir)
+            throws FileNotFoundException {
+
+        BufferedOutputStream out = null;
+        ZipInputStream in = null;
+
+        try {
+
+            in = new ZipInputStream(new BufferedInputStream(new FileInputStream(zipName)));
+            ZipEntry entry;
+            while ((entry = in.getNextEntry()) != null) {
+                int count;
+                byte data[] = new byte[1024];
+
+                out = new BufferedOutputStream(new FileOutputStream(outputDir.getPath() + "/" + entry.getName()), 1024);
+                while ((count = in.read(data, 0, 1024)) != -1) {
+                    out.write(data, 0, count);
+                }
+                out.flush();
+                IOUtils.closeQuietly(out, true);
+            }
+
+        } catch (Exception e) {
+            logger.error(e.getMessage(), e);
+        } finally {
+            IOUtils.closeQuietly(out, true);
+            IOUtils.closeQuietly(in, true);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

**Problem**
This line of code contains a resource that might not be closed properly. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *in*. The resource is referenced in statements at the following lines: 164, 166, 171, 179. The resource closure statement is at line: 182. There are other execution paths that do not contain closure statements, e.g., when exception is thrown by IOUtils.closeQuietly. Either a) close *in* in a try-finally block or b) close the resource by declaring *in* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).